src/logic/ast-analysis/typeInference.js

```diff
@@ -8,31 +9,46 @@ export default function assignFunctionTypes(
 ) {
   const resultAst = ast;
 
-  traverse.default(resultAst, {
-    FunctionDeclaration(path) {
-      path.node.params.forEach((param, index) => {
-        if (t.isIdentifier(param)) {
-          const inferredType = inferType(functionArguments[index]);
-          if (inferredType) {
-            param.typeAnnotation = t.tsTypeAnnotation(inferredType);
+  try {
+    traverse.default(resultAst, {
+      FunctionDeclaration(path) {
+        path.node.params.forEach((param, index) => {
+          if (t.isIdentifier(param)) {
+            const inferredType = inferType(functionArguments[index]);
+            if (inferredType) {
+              param.typeAnnotation = t.tsTypeAnnotation(inferredType);
+            }
           }
-        }
-      });
+        });
 
-      path.node.returnType = t.tsTypeAnnotation(inferType(userCodeResult));
-    },
+        const inferredType = inferType(userCodeResult);
+        if (inferredType) {
+          path.node.returnType = t.tsTypeAnnotation(inferredType);
+        } else {
+          path.node.returnType = t.tsTypeAnnotation(t.tsVoidKeyword());
+        }
+      },
 
-    VariableDeclarator(path) {
-      const { id, init } = path.node;
+      VariableDeclarator(path) {
+        const { id, init } = path.node;
 
-      if (t.isIdentifier(id)) {
-        const inferredType = inferType(init.value);
-        if (inferredType) {
-          id.typeAnnotation = t.tsTypeAnnotation(inferredType);
+        if (t.isIdentifier(id)) {
+          t.isArrayExpression(init);
+          const inferredType = inferType(init.value);
+          console.log(inferredType);
+          if (inferredType) {
+            id.typeAnnotation = t.tsTypeAnnotation(inferredType);
+          }
         }
-      }
-    },
-  });
+      },
+    });
+  } catch {
+    throw new ApiError(
+      "타입 추론 에러",
+      200,
+      "배열, 객체 타입은 지원하지 않습니다.",
+    );
```

Comment:
```suggestion
    const ERROR_CASE = {
        TYPE_INFER : {
            errorType: "타입 추론 에러",
            statusCode: 200,
            errorMessage: "배열, 객체 타입은 지원하지 않습니다.",
        }
    }
    
    throw new ApiError(ERROR_CASE.TYPE_INFER);
    );
```
현재 어떤 에러 타입인지, 상태코드, 에러메세지와 같은 부분들이 하드코딩으로 작성되어있습니다.

해당 하드코딩으로 작성되어 있는 부분을 `ERROR_CASE` 와 같은 에러 상수 객체를 정의하여 사용한다면 유지 보수 측면에서 효율적으로 사용할 수 있을 것으로 생각되는데요. 

다른 팀원님들과 리뷰어 님들의 의견은 어떠신가요?🤔